### PR TITLE
pusher: receipt can only be nil for ErrWantSelf cases

### PR DESCRIPTION
**Pull request by esadakar**

### CODE REVIEW

File: pkg/pusher/pusher.go [hunk 1]
```diff
@@ -165,21 +167,27 @@ LOOP:
...
 
-				storerPeer, err = crypto.NewOverlayAddress(*publicKey, s.networkID)
-				if err != nil {
-					err = fmt.Errorf("pusher: receipt storer address: %w", err)
+					storerPeer, err = crypto.NewOverlayAddress(*publicKey, s.networkID)
+					if err != nil {
+						err = fmt.Errorf("pusher: receipt storer address: %w", err)
+						return
+					}
+				} else if !wantSelf { // receipt is nil, so ErrWantSelf had to be returned
```
Comment (acud) [inline pkg/pusher/pusher.go:189]: not sure if it's necessary to check this. if I understand correctly, this case is `receipt == nil && !wantSelf` which shouldn't happen at all no?

Comment (esadakar) [reply pkg/pusher/pusher.go:189]: this is technically a part of the signature verification process. We should always get a receipt, and it should be signed. I think the added check is a good thing for cases where people run nodes with modified code bases. What do you think?

Comment (esadakar) [reply pkg/pusher/pusher.go:189]: ahh, the receipt is returned from PushChunkToClosest, the origin node constructs it from the stream, so yes, the extra check is not needed.

**Review by acud — APPROVED** [review]

**Review by ralph-pichler — APPROVED** [review]

**Pull request merged**